### PR TITLE
Add HashStringAllocator::Header::nextContinued API

### PR DESCRIPTION
**Pull request by mbasmanova**
Also, fix calculation of 'reserve' passed to HashStringAllocator::finishWrite in ValueList.

The existing calculation always returns 1024.

```
auto reserve = std::max<int32_t>(1024, std::min<int64_t>(128, totalBytes_));
```

The desired 'reserve' should be a value between 128 and 1024.

### CODE REVIEW

File: velox/common/memory/HashStringAllocator.h [hunk 1]
```diff
@@ -116,12 +117,21 @@ class HashStringAllocator : public StreamArena {
...
+    /// null if this is the last block of the arena.
     Header* FOLLY_NULLABLE next() {
       auto next = reinterpret_cast<Header*>(end());
       return next->data_ == kArenaEnd ? nullptr : next;
     }
 
+    /// Returns the header of the next block in a multi-part allocation. The
+    /// caller must ensure that isContinued() returns true before calling this
+    /// method.
+    HashStringAllocator::Header* FOLLY_NULLABLE nextContinued() {
```
Comment (xiaoxmeng) [inline velox/common/memory/HashStringAllocator.h:130]: Remove FOLLY_NULLABLE as it is not required any more?

File: velox/common/memory/HashStringAllocator.h [hunk 2]
```diff
@@ -50,6 +50,7 @@ class HashStringAllocator : public StreamArena {
     static constexpr uint32_t kContinued = 1U << 30;
     static constexpr uint32_t kPreviousFree = 1U << 29;
     static constexpr uint32_t kSizeMask = (1U << 29) - 1;
+    static constexpr uint32_t kContinuedOverhead = sizeof(void*);
```
Comment (xiaoxmeng) [inline velox/common/memory/HashStringAllocator.h:53]:
```
nit: s/kContinuedOverhead/kContinuedPointerSize or kContinuedPtrSize/?
```

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks!

File: velox/common/memory/HashStringAllocator.h (hunk 1 above)
Comment (yingsu00) [reply velox/common/memory/HashStringAllocator.h]: It prompts a warning in CLion by default so it's better to keep it

**Pull request closed without merging**